### ISSUE TITLE
quarterbacks and travel distance

### ISSUE THREAD

**Issue by easella** [CFBD/cfbd-python#8]
when you call get_games is there a way to get the name of quarterback of that game? What about travel distance?

**Comment by cooperjaXC** [comment]
Travel distance and travel method (e.g., bus, plane) would be very interesting.

**Comment by B-D-T** [comment]
# TLDR Recommend closing issue  

Great questions and ideas, but probably better suited for the [Discord](https://discord.com/channels/578652864793018368) than as a GitHub issue.

# Replies to OPs  

## Get QB name

> when you call get_games is there a way to get the name of quarterback of that game?  

@easella You could combine your game data with the [getadvancedboxscore](https://api.collegefootballdata.com/api/games#getadvancedboxscore) data to see which QB for each team took the most snaps. Or, to get starters, combine with [plays](https://api.collegefootballdata.com/api/plays#getplays) to see which QB took the first snaps for each team.


## Travel distance   

> Travel distance and travel method (e.g., bus, plane) would be very interesting.  

@cooperjaXC Unfortunately, there's no clear source for bus vs plane. You can get close through basic logic, reviewing promotional content, and even some FOIA requests, but at some point, it's just an educated guess.[^1]  

Anyway, there are a few ways to get the [geographic distance](https://en.wikipedia.org/wiki/Geographical_distance) between two points. I recommend Karney's ([2013](https://link.springer.com/article/10.1007/s00190-012-0578-z)) calculation, but that's mostly because it's what [geopy](https://geopy.readthedocs.io/en/stable/#module-geopy.distance) uses in its [distance.geodesic](https://geopy.readthedocs.io/en/stable/#geopy.distance.geodesic) calculation. 

The [CFBDb games api endpoint](https://api.collegefootballdata.com/api/games) includes the `venue_id`, and each venue has the lat/lon. Then, combine that with the [teams endpoint](https://api.collegefootballdata.com/api/teams) to get the distances traveled by the home/away teams.

[^1]: Coincidently, I'm doing research on this right now. Well, actually I'm getting sucked in by playing with the data, but don't tell my co-authors.

**Comment by cooperjaXC** [comment]
@B-D-T, thanks for your comments and the Karney recommendation. I have had good luck with using `geopy` and `geopandas` for calculating Euclidian distances, as well as PostGIS. 

Good luck on your research! I have done some for [college football](https://doi.org/10.1080/14775085.2020.1726802) and [basketball](https://doi.org/10.1080/14775085.2020.1726802) myself where, as you suggest, I made educated guesses on travel mode. Also, if you want to move beyond straight-line distances to model transportation routes, I have tinkered around with some lightweight network analysis [python tools](https://github.com/ccaoa/osmh3nx) that might be of use to you. It's in its infancy and needs more attention (that I'm not able to give it right now). Looking forward to your results!
